client/src/components/UploadMusic/MusicUpload.jsx

```diff
@@ -36,17 +37,32 @@ const MusicUpload = () => {
     widget.open();
   };
 
-  useEffect( () => {
+  const getMusic = async () => {
+    try {
+      const music = await axios.get('/upload/musicUpload');
+      console.info('Music??');
+      setMusic(music.data[0].MusicUploads);
+    } catch (err) {
+      console.warn(err);
+    }
+  };
 
-    const getMusic = async () => {
-      try {
-        const music = await axios.get('/upload/musicUpload');
-        setMusic(music.data[0].MusicUploads);
-      } catch (err) {
-        console.warn(err);
-      }
-    };
-    getMusic();
+  useEffect(() => {
+
+    if (currentUser) {
+      getMusic();
+    }
+   
+  }, [currentUser]);
+
+  
+  
+
+  useEffect(() => {
```

Comment:
are you sure all errors are gone?  before this was causing problems when deployed and crashing the app.  It is probably more efficient to get this info from the context.